.github/workflows/publish-cli-bins.yaml

```diff
@@ -9,89 +9,160 @@ on:
   pull_request:
 
 jobs:
+  setup:
+    name: Set Up Shared Environment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y make
+
   build:
+    name: Build Binaries
     strategy:
       matrix:
-        os: [linux, darwin]
-        arch: [amd64, arm64]
-    env:
-      NAME: flightctl-${{ matrix.os }}-${{ matrix.arch }}
-    runs-on: "ubuntu-24.04"
+        os_arch: [ "linux-amd64", "linux-arm64", "darwin-amd64", "darwin-arm64" ]
+    runs-on: ubuntu-latest
+    needs: setup
     steps:
-      - name: "Checkout"
+      - name: Set up Environment Variables
+        run: |
+          # Split os-arch into os and arch
+          IFS="-" read -r GOOS GOARCH <<< "${{ matrix.os_arch }}"
+          # Export variables to GITHUB_ENV for subsequent steps
+          {
+            echo "GOOS=$GOOS"
+            echo "GOARCH=$GOARCH"
+          } >> $GITHUB_ENV
+
+      - name: Checkout Code
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-      - name: "Build"
-        env:
-          GOARCH: ${{ matrix.arch }}
-          GOOS: ${{ matrix.os }}
+      - name: Cache Git Repository
+        uses: actions/cache@v4
+        with:
+          path: .git
+          key: ${{ runner.os }}-git-${{ hashFiles('**/.git') }}
+          restore-keys: |
+            ${{ runner.os }}-git-
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.21
+
+      # Cache Go modules
+      - name: Cache Go Modules
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      # Install dependencies
+      - name: Install Dependencies
+        run: go mod tidy
+
+      - name: Build
         run: |
-          make build-cli
+          GOOS="${GOOS}" GOARCH="${GOARCH}" make build-cli
```

Comment:
What I was proposing was more around having a make build-cli-multiarch target itself that you could use from anywhere

inside the target you could loop. or have a script that runs

for GOARCH in amd64 arm64; do
     for GOOS in linux darwin windows; do
            GOOS="${GOOS}" GOARCH="${GOARCH}" make build-cli
            # SHA signature
            # renaming binaries
      done
done

But put this acessible from a make target.


The reason for asking this, is that then we can use this in a Container file to build the "container wrapped" cli binaries.